libraries/Microsoft.Bot.Builder.Skills/SkillAdapter.cs

```diff
@@ -0,0 +1,416 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Bot.Builder.Skills
+{
+    /// <summary>
+    /// A Akill adapter implements API to forward activity to a skill and routes ChannelAPI 
+    /// calls from the Skill up through the bot/adapter.
+    /// </summary>
+    public abstract class SkillAdapter
+    {
+        private readonly ILogger logger;
+
+        public SkillAdapter(BotAdapter adapter, IBot bot, ILogger logger = null)
+        {
+            this.ChannelAdapter = adapter;
+            this.Bot = bot;
+            this.logger = logger ?? NullLogger.Instance;
+
+            // make sure there is a channel api middleware
+            if (!this.ChannelAdapter.MiddlewareSet.Any(mw => mw is ChannelApiMiddleware))
+            {
+                this.ChannelAdapter.MiddlewareSet.Use(new ChannelApiMiddleware(this));
+            }
+        }
```

Comment:
We shouldn't be adding this automatically.  What if they want a different implementation with different policies?

Reply:
This is internal only for now, we can open it up later once we find a use case for it. 
Otherwise the developer will need to add it and we will need to make it public (and I think we are still not sold on the idea of using a middleware, right?)